src/unity/toolkits/object_detection/object_detector.cpp

```diff
@@ -319,8 +319,26 @@ variant_map_type object_detector::evaluate(
     gl_sframe data, std::string metric,
     std::map<std::string, flexible_type> opts) {
 
-  // TODO: Support user-configurable metric and options.
-
+  std::vector<std::string> metrics;
+  static constexpr char AP[] = "average_precision";
+  static constexpr char MAP[] = "mean_average_precision";
+  static constexpr char AP50[] = "average_precision_50";
+  static constexpr char MAP50[] = "mean_average_precision_50";
+  std::vector<std::string> all_metrics = {AP,MAP,AP50,MAP50};
```

Comment:
Let's go ahead and leave `all_metrics` default-initialized in this line:
```
std::vector<std::string> all_metrics;
```
Since you already handle this case in line 335.